<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>resume</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <style type="text/css">
  /*
   * Copyright 2013 Christophe-Marie Duquesne <chmd@chmd.fr>
   *
   * CSS for making a resume with pandoc. Inspired by moderncv.
   *
   * This CSS document is delivered to you under the CC BY-SA 3.0 License.
   * https://creativecommons.org/licenses/by-sa/3.0/deed.en_US
   */
  
  /* Whole document */
  body {
    {% include analytics/analytics.html %}

      font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
      max-width: 800px;
      margin: auto;
      background: #FFFFFF;
      padding: 10px 10px 10px 10px;
  }
  
  /* Title of the resume */
  h1 {
      font-size: 55px;
      color: #757575;
      text-align:center;
      margin-bottom:15px;
  }
  h1:hover {
      background-color: #757575;
      color: #FFFFFF;
      text-shadow: 1px 1px 1px #333;
  }
  
  /* Titles of categories */
  h2 {
      /* This is called "sectioncolor" in the ConTeXt stylesheet. */
      color: #b7709c;
  }
  /* There is a bar just before each category */
  h2:before {
      content: "";
      display: inline-block;
      margin-right:1%;
      width: 16%;
      height: 10px;
      /* This is called "rulecolor" in the ConTeXt stylesheet. */
      background-color: #709cb7;
  }
  h2:hover {
      background-color: #397249;
      color: #FFFFFF;
      text-shadow: 1px 1px 1px #333;
  }
  
  /* Definitions */
  dt {
      float: left;
      clear: left;
      width: 17%;
      /*font-weight: bold;*/
  }
  dd {
      margin-left: 17%;
  }
  p {
      margin-top:0;
      margin-bottom:7px;
  }
  
  /* Blockquotes */
  blockquote {
      text-align: center
  }
  
  /* Links */
  a {
      text-decoration: none;
      color: #397249;
  }
  a:hover, a:active {
      background-color: #397249;
      color: #FFFFFF;
      text-decoration: none;
      text-shadow: 1px 1px 1px #333;
  }
  
  /* Horizontal separators */
  hr {
      color: #8b70b7;
  }
  
  table {
      width: 100%;
  }
  </style>
</head>
<body>
<h1 id="michel-moser-phd">Michel Moser, PhD</h1>
<blockquote>
<p><a href="mailto:michel.moser@ips.unibe.ch">michel.moser@ips.unibe.ch</a> • +41 (0)79 744 6066 • michel_moser<br />
<a href="https://michelmoser.github.io" target="_blank">https://michelmoser.github.io/</a></p>
</blockquote>
<table>
<tbody>
<tr class="odd">
<td><strong>Position:</strong>   Post-doctoral researcher</td>
<td><strong>Address:</strong>  Michel Moser   </td>
</tr>
<tr class="even">
<td>      Kuhlemeier lab</td>
<td>     Stationsstrasse 37</td>
</tr>
<tr class="odd">
<td><strong>Institution:</strong>  Institute of Plant Sciences</td>
<td>     3645 Gwatt, Bern</td>
</tr>
<tr class="even">
<td>      University of Bern</td>
<td><strong>Birth Date:</strong> 17.01.1984</td>
</tr>
</tbody>
</table>
<hr />
<h2 id="education">Education</h2>
<dl>
<dt>07/2013 - 01/2018</dt>
<dd><p><strong>PhD, Science in Cell Biology</strong>; University of Bern (Bern)</p>
<p>Thesis title: <em>Speciation genomics in Petunia</em><br />
Main focus: <em>Large-scale genomic data mining, genome assembly</em><br />
Grade: <em>Insigni cum laude</em></p>
</dd>
<dt>07/2011 - 06/2013</dt>
<dd><p><strong>MSc, Science in Molecular Life Sciences</strong>; University of Bern (Bern)</p>
<p>Thesis title: <em>RNA-seq analysis of floral tissue from Petunia</em><br />
Main focus: <em>Analysis of next generation sequencing data</em><br />
Grade: <em>Summa cum laude</em></p>
</dd>
<dt>09/2008 - 07/2011</dt>
<dd><p><strong>BSc, Science in Biology</strong>; University of Bern (Bern)</p>
<p>Thesis title: <em>“Interaktionstest und Lokalisierung von AtPTR2 und AtWLIM2b”</em><br />
Grade: <em>Insigni cum laude</em></p>
</dd>
<dt>09/2007 - 07/2008</dt>
<dd><strong>Qualification for University Entrance</strong>; Gymnasium Neufeld (Bern)
</dd>
<dt>08/2003 - 07/2006</dt>
<dd><strong>Technical Berufsmaturitaet</strong>; GIBB (Bern)
</dd>
<dt>08/2003 - 07/2006</dt>
<dd><strong>Federal Diploma of Higher Education in Landscape Gardening</strong>; Paradies Gaerten
</dd>
<dt>08/2001 - 07/2003</dt>
<dd><strong>High school</strong>; Gymnasium Schadau (Thun)
</dd>
</dl>
<h2 id="work-experience">Work Experience</h2>
<dl>
<dt>02/2018 - now</dt>
<dd><strong>Postdoc in Bioinformatics and Computational Biology</strong>; University of Bern
</dd>
<dd>Computational plant genomics research including: genome assembly and annotation, genetic map construction, comparative genomics, speciation divergence analyses, development of computational pipelines to detect allele-specific expression
</dd>
<dd>Assistance and guidance of Bachelor and Master student projects
</dd>
<dt>07/2006 - 08/2007</dt>
<dd><strong>Landscape gardener</strong>; Paradies Gaerten, Bern
</dd>
</dl>
<p><strong>Part-time jobs:</strong></p>
<dl>
<dt>08/2017 - now</dt>
<dd><strong>Data scientist</strong>; whoch2 Gmbh, Kramgasse 5, Bern
</dd>
<dd>External advisor, methodically assistance in data analysis and visualization, trained random forest model to predict costumer behaviour
</dd>
<dd><a href="www.w-hoch2.ch">w-hoch2.ch</a>
</dd>
<dt>07/2013 - 01/2018</dt>
<dd><strong>Research assistant, 50 %</strong>; University of Bern
</dd>
<dd>Completion of the first genome assembly of Petunia
</dd>
<dd>Contribution to the identification of genes responsible for UV aborbance
</dd>
<dt>08/2007 - 09/2008</dt>
<dd><strong>Landscape gardener, 60 %</strong>; Paradies Gaerten, Bern
</dd>
</dl>
<h2 id="skills">Skills</h2>
<p><strong>Soft skills</strong></p>
<p>Interacted with many reserachers from diverse fields for co-authored research work. Delivered presentations of research work at several conferences and at group and institute-wide meetings. Extensive exchange about methods and technology</p>
<p><strong>Software engineering</strong></p>
<dl>
<dt>OS</dt>
<dd>Proficient in Linux/Unix, Microsoft Windows
</dd>
<dt>Programming Languages</dt>
<dd>python: pandas, scikit-learn, seaborn
</dd>
<dd>R: ggplot2, dplyr, caret, randomforest, e1071, reshape2
</dd>
<dd>others: bash, awk, sed, vim, java
</dd>
<dt>Databases</dt>
<dd>MySQL, PostgresSQL
</dd>
</dl>
<p><strong>Bioinformatics</strong></p>
<p>Extensive working knowledge in anaylsis of genomics data using R/Bioconductor, Linux shell scripting and python</p>
<ul>
<li>mining and analysis of large sequencing datasets (&gt;100 GBytes)</li>
<li>use of high performance computing clusters</li>
<li>sequence assembly and alignment using state-of-the-art software</li>
<li>sequence-based clustering (PCA, MDS)</li>
<li>implementation of computational pipelines (sliding-window approaches, genetic marker detection, allele-specific expression)</li>
<li>imputation of missing genetic markers using random forest</li>
<li>training of Hidden Markov Models for gene modelling
<ul>
<li>implementation of web-crawler to fetch genomic sequence information</li>
</ul></li>
</ul>
<dl>
<dt>Statistics</dt>
<dd>Good understanding of biostatistical methodology and tools
</dd>
</dl>
<p><strong>Languages</strong> German (mother tongue), English (very good), French (basic)</p>
<h2 id="publications">Publications</h2>
<dl>
<dt>2016</dt>
<dd>Gain and loss of floral scent production through changes in structural genes during pollinatior-mediated speciation.
</dd>
<dd>Amrad, A., Moser, M., Mandel, T., De Vries, M.,.. &amp; Kuhlemeier, C.
</dd>
<dd><a href="https://www.sciencedirect.com/science/article/pii/S0960982216312088?via%3Dihub" target="_blank"><em>Current Biology</em></a>
</dd>
<dt>2016</dt>
<dd>Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida.
</dd>
<dd>Moser, M., Bombarely, A., Amrad, A., Bao, M.,.. &amp; Kuhlemeier, C.
</dd>
<dd><a href="https://www.nature.com/articles/nplants201674https://www.nature.com/articles/nplants201674" target="_blank"><em>Nature plants</em></a>
</dd>
<dt>2015</dt>
<dd>MYB-FL controls gain and loss of floral UV absorbance, a key trait affecting pollinator preference and reproductive isolation.
</dd>
<dd>Sheehan, H., Moser, M., Klahre, U., Esfeld, K., .. &amp; Kuhlemeier, C.
</dd>
<dd><a href="https://www.nature.com/articles/ng.3462" target="_blank"><em>Nature genetics</em></a>
</dd>
<dt>2013</dt>
<dd>Tight genetic linkage of prezygotic barrier loci creates a multifunctional speciation island in Petunia.
</dd>
<dd>Hermann, K., Klahre, U., Moser, M., Sheehan, H., Mandel, T., &amp; Kuhlemeier, C.
</dd>
<dd><a href="https://www.sciencedirect.com/science/article/pii/S0960982213003710?via%3Dihub" target="_blank"><em>Current Biology</em></a>
</dd>
</dl>
<h2 id="interests">Interests</h2>
<ul>
<li><p>Trail running</p></li>
<li><p>Mountaineering</p></li>
<li>member of the Academic Alpine Club Bern *member of the Academic Alpine Club Bern</li>
<li><p>cashier of the Academic Alpine Club Bern</p></li>
</ul>
<hr />
<blockquote>
<p><a href="mailto:michel.moser@ips.unibe.ch">michel.moser@ips.unibe.ch</a> • +41 (0)79 744 6066 • michel_moser<br />
<a href="https://michelmoser.github.io" target="_blank">https://michelmoser.github.io/</a></p>
</blockquote>
</body>
</html>
